classifier/rbdigital.py

```diff
@@ -0,0 +1,97 @@
+from . import *
+
+class RBDigitalAudienceClassifier(FreeformAudienceClassifier):
+
+    @classmethod
+    def target_age(cls, identifier, name):
+        # RBdigital uses 'beginning reader' to refer to books
+        # suitable from birth to age 8, rather than (as is more common)
+        # ages 5 to 8.
+        #
+        # Rather than covering the entire early lifespan of a child,
+        # the normally vague 'childrens' is used here to cover the
+        # time in between 'beginning reader' and 'young adult'
+        if identifier == 'beginning reader':
+            return cls.range_tuple(0,8)
+        elif identifier == 'childrens':
+            return cls.range_tuple(9, 13)
+        return FreeformAudienceClassifier.target_age(identifier, name)
+
+class RBDigitalSubjectClassifier(KeywordBasedClassifier):
+
+    # We have subgenres for fiction in these categories but not for
+    # nonfiction in these categories (thus the None mapping in
+    # 'genres' below).
+    fiction_genres = {
+        'lgbt interest' : LGBTQ_Fiction,
+    }
+
+    genres = {
+        # This isn't true in general but because RBdigital is heavy on
+        # audio content, 'arts and entertainment' is more likely to be
+        # entertainment (e.g. music) than arts.
+        'arts entertainment' : Entertainment,
+        'business economics' : Business,
+        'comics graphic novels' : Comics_Graphic_Novels,
+        'home garden' : Hobbies_Home,
+        # Determined by looking at a sample collection.
+        'humor' : Humorous_Nonfiction,
+
+        # We don't check this in KeywordBasedClassifier to avoid
+        # confusion with e.g. 'western civ'.
+        'western' : Westerns,
+
+        # If we go to this point we know it's not fiction so we have
+        # nothing to say about the genre -- it could be anything.
+        'lgbt interest' : None,
+    }
+
+    @classmethod
+    def scrub_identifier(cls, identifier):
+        return identifier.replace('-', ' ')
+
+    @classmethod
+    def genre(cls, identifier, name, fiction=None, audience=None, **kwargs):
+        if fiction and identifier in cls.fiction_genres:
+            return cls.fiction_genres[identifier]
+        if identifier in cls.genres:
+            return cls.genres[identifier]
+        return KeywordBasedClassifier.genre(
+            identifier, name, fiction, audience, **kwargs
+        )
+
+    @classmethod
+    def audience(cls, identifier, name):
+        """The two subjects here that imply an audience, juvenile-fiction and
+        juvenile-nonfiction, could be either Childrens or YA, and
+        RBdigital always provides a more specific audience
+        classification, so we don't try to derive audience information
+        from an RBDigital Subject.
+        """
+        if identifier == 'erotica':
+            return Classifier.AUDIENCE_ADULTS_ONLY
```

Comment:
This isn't necessary--I didn't even see any "erotica" in my RBdigital collection and it'll be handled by the caller anyway--but just to be double-safe.